remill/Arch/AArch64/Arch.cpp

```diff
@@ -719,6 +719,28 @@ static bool DecodeBitMasks(uint64_t N /* one bit */,
   }
   return true;
 }
+/// Utility function for extracting [From, To] bits from a uint32_t.
+static inline unsigned slice(uint32_t Bits, unsigned From, unsigned To) {
+  assert(From < 32 && To < 32 && From >= To);
+  return (Bits >> To) & ((1 << (From - To + 1)) - 1);
+}
+
+// Adapted from LLVM 
+// https://llvm.org/svn/llvm-project/llvm/branches/ggreif/CallInst-operands/lib/Target/ARM/Disassembler/ARMDisassemblerCore.cpp
+static uint64_t VFPExpandImm(uint8_t imm, RegClass float_class) {
+  assert(float_class == kRegS || float_class == kRegD || float_class == kRegH);
+  uint64_t result;
+
+  unsigned bit6 = slice(imm, 6, 6);
+  if(kRegD == float_class) {
+    result = (uint64_t)slice(imm, 7, 7) << 63 | (uint64_t)slice(imm, 5, 0) << 48;
+    bit6? result |= 0xffULL << 54l : result |= 0x1ULL << 62;
+  } else {
+    result = slice(imm, 7, 7) << 31 | slice(imm, 5, 0) << 19;
+    bit6? result |= 0x1f << 25 : result |= 0x1 << 30;
```

Comment:
Space before the `?`.

Reply:
Break this out into an `if` statement to avoid putting these assignments inside the ternary expression.